packages/webapp/pages/__tests__/callback.spec.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import type { RenderResult } from '@testing-library/react';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import * as func from '@dailydotdev/shared/src/lib/func';
+import { AuthEvent } from '@dailydotdev/shared/src/lib/kratos';
+import CallbackPage from '../callback';
+
+/**
+ * Extend Jest matchers
+ */
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace jest {
+    interface Matchers<R> {
+      toBeNull(): R;
+    }
+  }
+}
+
+const mockLogEvent = jest.fn();
+
+jest.mock('@dailydotdev/shared/src/contexts/LogContext', () => ({
+  __esModule: true,
+  default: {
+    Consumer: jest.fn(),
+    Provider: jest.fn(),
+  },
+  LogContextProvider: ({ children }: { children: React.ReactNode }) => children,
+}));
+
+// Create mock context
+const MockLogContext = React.createContext({ logEvent: mockLogEvent });
+
+// Mock the window location and opener
+const mockReplace = jest.fn();
+const mockClose = jest.fn();
+
+describe('CallbackPage', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    // Setup window mocks
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    delete (window as any).location;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (window as any).location = { replace: mockReplace, search: '' };
```

Comment:
Deleting this looks hacky, something like we do for `matchMedia` in other tests sounds better, we can also restore real version in `afterEach`